packages/share/android/src/main/java/io/flutter/plugins/share/Share.java

```diff
@@ -47,4 +61,150 @@ void share(String text, String subject) {
       activity.startActivity(chooserIntent);
     }
   }
+
+  void shareFiles(List<String> paths, List<String> mimeTypes, String text, String subject)
+      throws IOException {
+    if (paths == null || paths.isEmpty()) {
+      throw new IllegalArgumentException("Non-empty path expected");
+    }
+
+    clearExternalShareFolder();
+    ArrayList<Uri> fileUris = getUrisForPaths(paths);
+
+    Intent shareIntent = new Intent();
+    if (fileUris.size() == 1) {
+      shareIntent.setAction(Intent.ACTION_SEND);
+      shareIntent.putExtra(Intent.EXTRA_STREAM, fileUris.get(0));
+      shareIntent.setType(
+          !mimeTypes.isEmpty() && mimeTypes.get(0) != null ? mimeTypes.get(0) : "*/*");
+    } else {
+      shareIntent.setAction(Intent.ACTION_SEND_MULTIPLE);
+      shareIntent.putParcelableArrayListExtra(Intent.EXTRA_STREAM, fileUris);
+      shareIntent.setType(reduceMimeTypes(mimeTypes));
+    }
+    if (text != null) shareIntent.putExtra(Intent.EXTRA_TEXT, text);
+    if (subject != null) shareIntent.putExtra(Intent.EXTRA_SUBJECT, subject);
+    shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+    Intent chooserIntent = Intent.createChooser(shareIntent, null /* dialog title optional */);
+
+    List<ResolveInfo> resInfoList =
+        activity
+            .getPackageManager()
+            .queryIntentActivities(chooserIntent, PackageManager.MATCH_DEFAULT_ONLY);
+    for (ResolveInfo resolveInfo : resInfoList) {
+      String packageName = resolveInfo.activityInfo.packageName;
+      activity.grantUriPermission(
+          packageName,
+          fileUris.get(0),
+          Intent.FLAG_GRANT_WRITE_URI_PERMISSION | Intent.FLAG_GRANT_READ_URI_PERMISSION);
+    }
+
+    if (activity != null) {
+      activity.startActivity(chooserIntent);
+    } else {
+      chooserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+      activity.startActivity(chooserIntent);
```

Comment:
Good point :). Seems like this was code that I copied over from master. The already existing `share()` method has the same. But agree this would crash. I changed the code now to handle this better and even just use the application context if no activity is attached.